### PR TITLE
Expose scipp.and and scipp.all for all dimensions

### PR DESCRIPTION
**Pull request by OwenArnold**
Brings these functions inline with `max` and `min` of reduction in terms of behaviour.

### CODE REVIEW

File: python/reduction.cpp [hunk 1]
```diff
@@ -165,24 +165,42 @@ template <class T> Docstring docstring_bool(const std::string op) {
...
+template <class T> Docstring docstring_bool(const std::string op) {
+  return Docstring()
+      .description("Element-wise " + op +
+                   " over all of the input's dimensions.")
+      .raises("If the dtype has no " + op + ", e.g., if it is a string.")
+      .returns("The " + op + " combination of the input values.")
+      .rtype<T>()
+      .template param<T>("x", "Data to reduce.");
+}
+
```
Comment (SimonHeybrock) [inline python/reduction.cpp:177]: I suggest to use the same docstring for both and write something like: `.param("dim", "Dimension to reduce (optional).", "Dim");`, and not in the `description` that all dimensions are reduced unless `dim` is given.

File: python/reduction.cpp [hunk 2]
```diff
@@ -157,25 +157,36 @@ template <class T> void bind_max(py::module &m) {
...
+      "all", [](const typename T::const_view_type &x) { return all(x); },
+      py::arg("x"), py::call_guard<py::gil_scoped_release>(),
+      docstring_bool<T>("AND").c_str());
   m.def(
       "all",
       [](const typename T::const_view_type &x, const Dim dim) {
         return all(x, dim);
       },
-      py::arg("x"), py::arg("dim"), py::call_guard<py::gil_scoped_release>(),
+      py::arg("all"), py::arg("dim"), py::call_guard<py::gil_scoped_release>(),
```
Comment (nvaytet) [inline python/reduction.cpp:181]: Why this change?
Isn't it strange to have an argument which is the same name as the function itself?
It is also no longer in agreement with the version without the `dim` just above.
